Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -778,7 +778,17 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
       )
```

Comment:
I also discovered issues with the C-style for loop, but it seemed difficult to resolve them simply 🤯
It appears that they should be considered as separate issues.